packages/suite/src/actions/wallet/blockchainActions.ts

```diff
@@ -262,39 +271,58 @@ export const unsubscribe = (removedAccounts: Account[]) => (_: Dispatch, getStat
+    if (timeout) clearTimeout(timeout);
+};
+
+export const syncAccounts =
+    (symbol: Network['symbol']) => async (dispatch: Dispatch, getState: GetState) => {
+        // First clear, to cancel last planned sync
+        tryClearTimeout(getState().wallet.blockchain[symbol].syncTimeout);
+
+        await Promise.all(
+            getState()
+                .wallet.accounts.filter(a => a.symbol === symbol)
+                .map(a => dispatch(accountActions.fetchAndUpdateAccount(a))),
+        );
+
+        // Second clear, just to be sure that no other sync was planned while executing this one
+        tryClearTimeout(getState().wallet.blockchain[symbol].syncTimeout);
+        const timeout = setTimeout(() => dispatch(syncAccounts(symbol)), ACCOUNTS_SYNC_INTERVAL);
+
+        dispatch({
+            type: BLOCKCHAIN.SYNCED,
+            payload: {
+                symbol,
+                timeout,
+            },
+        });
+    };
+
 export const onConnect = (symbol: string) => async (dispatch: Dispatch, getState: GetState) => {
     const network = getNetwork(symbol.toLowerCase());
     if (!network) return;
     const blockchain = getState().wallet.blockchain[network.symbol];
-    if (blockchain.reconnection) {
-        // reset previous timeout
-        clearTimeout(blockchain.reconnection.id);
-    }
+    // reset previous timeout
+    tryClearTimeout(blockchain.reconnection?.id);
     await dispatch(subscribe(network.symbol, true));
     await dispatch(updateFeeInfo(network.symbol));
     // update accounts for connected network
-    const accounts = getState().wallet.accounts.filter(a => a.symbol === network.symbol);
-    if (accounts.length) {
-        const promises = accounts.map(a => dispatch(accountActions.fetchAndUpdateAccount(a)));
-        await Promise.all(promises);
-    }
+    await dispatch(syncAccounts(network.symbol));
     dispatch({ type: BLOCKCHAIN.CONNECTED, payload: symbol });
 };
 
-export const onBlockMined =
-    (block: BlockchainBlock) => (dispatch: Dispatch, getState: GetState) => {
-        const symbol = block.coin.shortcut.toLowerCase();
-        const networkAccounts = getState().wallet.accounts.filter(a => a.symbol === symbol);
-        const promises = networkAccounts.map(account =>
-            dispatch(accountActions.fetchAndUpdateAccount(account)),
-        );
-        return Promise.all(promises);
-    };
+export const onBlockMined = (block: BlockchainBlock) => (dispatch: Dispatch) => {
+    const symbol = block.coin.shortcut.toLowerCase();
+    if (!accountUtils.isNetworkSymbol(symbol)) return;
+    return dispatch(syncAccounts(symbol as Network['symbol']));
+};
 
 export const onNotification =
     (payload: BlockchainNotification) => (dispatch: Dispatch, getState: GetState) => {
         const { descriptor, tx } = payload.notification;
         const symbol = payload.coin.shortcut.toLowerCase();
+        if (!accountUtils.isNetworkSymbol(symbol)) return;
```

Comment:
Because the symbol here comes from the `TrezorConnect`, so its type is string although it should always be one of known coins. `isNetworkSymbol` typeguard is definitely safer than `as` keyword in cases like this.

Reply:
thanks :)